src/snap/metric.cc

```diff
@@ -263,3 +263,111 @@ void Metric::Metric::set_last_advert_tm(system_clock::time_point tp) {
     tm->set_sec(nanos / std::nano::den);
     tm->set_nsec(nanos % std::nano::den);
 }
+
+Namespace::Namespace(std::vector<std::string> ns){
+    
+    for (auto &string_iterator : ns){
+        this->namespace_elements.push_back(NamespaceElement(string_iterator));
+    }
+}
+Namespace::Namespace(){}
```

Comment:
Please add empty line before this constructor to have clear formatting in code which you have added.